pintc/src/asm_gen.rs

```diff
@@ -103,6 +104,38 @@ pub fn compile_contract(
         }
     }
 
+    // check for duplicate predicates
+    let mut unique_addresses: FxHashSet<&ContentAddress> = FxHashSet::default();
+    let mut original_predicate_names: FxHashMap<&ContentAddress, &String> = FxHashMap::default();
+    for (string, (_, content_address)) in &compiled_predicates {
+        if !unique_addresses.insert(content_address) {
+            let original_name = original_predicate_names
+                .get(content_address)
+                .expect("predicate name guaranteed to exist");
+            let original_span = contract
+                .symbols
+                .symbols
+                .get(*original_name)
+                .expect("original predicate name missing in contract symbols table");
+            let span = contract
+                .symbols
+                .symbols
+                .get(string)
+                .expect("predicate name missing in contract symbols table");
```

Comment:
This really is a pain.. Starting to think @otrho's suggestion is better. So yeah you'd have to change the parser to collect an `Ident` for `Predicate` instead of a `String`. Then, the span is readily available through `.span`.

Reply:
https://github.com/essential-contributions/pint/blob/04094621349fcf42e31bb6e8bee053f66f7f07c4/pintc/src/predicate.rs#L956-L961

Is this what you had in mind? 

it results in a lot of `pred.name.name` everywhere that used to use a `string`. Maybe not an issue?

Or did you want me to stuff span in to the struct itself ala 
```
pub struct Predicate {
    pub name: String,
    pub span: Span,
    ...
```